### PR TITLE
prune locks map in gstd

### PR DESCRIPTION
**Pull request by grishasobol**
without this patch locks map will grows forever for any contract with asyncs 

Change internal map for locks back to BTreeMap, because I think is more probably, that internal map for messages is small, so it's more effective to use BTreeMap than HashMap for small data

### CODE REVIEW

File: gstd/src/lib.rs [hunk 1]
```diff
@@ -31,6 +31,8 @@
 #[cfg(target_arch = "wasm32")]
 extern crate galloc;
 
+extern crate alloc;
```
Comment (breathx) [inline gstd/src/lib.rs:34]: alloc and btreemap is already exported in prelude mod

File: gstd/src/async_runtime/locks.rs [hunk 2]
```diff
@@ -158,15 +157,13 @@ impl LocksMap {
         // Locks with `deadline < now`, they should already been removed since
         // the node will trigger timeout when locks reaching their deadline.
         //
-        // Locks with `deadline == now`, they will be removed in the following
-        // pollings.
+        // Locks with `deadline == now`, they will be removed in the following pollings.
```
Comment (breathx) [inline gstd/src/async_runtime/locks.rs:160]:
```suggestion
        // Locks with `deadline <= now`, they will be removed in the following polling.
```

File: gstd/src/async_runtime/futures.rs [hunk 3]
```diff
@@ -68,6 +68,7 @@ where
 
     if Pin::new(&mut task.future).poll(&mut cx).is_ready() {
         super::futures().remove(&msg_id);
+        super::locks().remove_message_entry(msg_id);
```
Comment (breathx) [inline gstd/src/async_runtime/futures.rs:71]: One more place to remove lock
https://github.com/gear-tech/gear/blob/7c0ae01484e77bf4d5f5d7dcd830829b68011263/gstd/src/async_runtime/mod.rs#L58-L60

File: gstd/src/async_runtime/locks.rs [hunk 4]
```diff
@@ -181,6 +178,16 @@ impl LocksMap {
...
     }
 
+    pub fn remove_message_entry(&mut self, message_id: MessageId) {
+        self.0.remove(&message_id);
+        // Question for reviewers: how map for message can be not empty at this
+        // point ?!!! .map(|locks| {
+        //     if !locks.is_empty() {
+        //         // unreachable!("Wanna to remove locks map for message, but
+        // the map is not empty: {:?}", locks);     }
+        // });
```
Comment (breathx) [inline gstd/src/async_runtime/locks.rs:188]: It can't so the panic on `None` removal res seems ok

**Review by breathx — CHANGES_REQUESTED** [review]

File: gstd/src/lib.rs (hunk 1 above)
Comment (grishasobol) [reply gstd/src/lib.rs:34]: ok removed

File: gstd/src/async_runtime/locks.rs (hunk 2 above)
Comment (grishasobol) [reply gstd/src/async_runtime/locks.rs]: done

File: gstd/src/async_runtime/futures.rs (hunk 3 above)
Comment (grishasobol) [reply gstd/src/async_runtime/futures.rs:71]: added

File: gstd/src/async_runtime/locks.rs (hunk 4 above)
Comment (grishasobol) [reply gstd/src/async_runtime/locks.rs:187]: This panic is appeared some times, so we should discuss it more

Comment (breathx) [reply gstd/src/async_runtime/locks.rs:187]: discussed

**Review by breathx — APPROVED** [review]

**Pull request merged**